.github/workflows/PublishNext.yml

```
@@ -6,20 +6,48 @@ on:
     - cron: '0 7 * * *'
 
 jobs:
-  publish:
+  check:
     if: github.repository == 'roblox-ts/roblox-ts' && github.ref == 'refs/heads/master'
     runs-on: ubuntu-latest
+    outputs:
+      version: ${{ env.VERSION }}
+      version_changed: ${{ env.VERSION_CHANGED }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set version
+        run: echo "VERSION=$(cat package.json | jq -r .version)-dev-$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+
+      - name: Check version
+        run: |
+            LATEST_VERSION=$(curl --silent https://registry.npmjs.org/roblox-ts | jq -r '."dist-tags" | .next')
```

I think we can simplify this by using `npm view`.

```suggestion
            LATEST_VERSION=$(npm view roblox-ts dist-tags.next)
```